src/main/resources/avro/bdg.avdl

```diff
@@ -988,6 +703,97 @@ record VariantAnnotation {
 }
 
 /**
+ Allele encodings for genotypes.
+ */
+enum Allele {
+
+  /**
+   Reference allele.
+   */
+  REF,
+
+  /**
+   Alternate allele.
+   */
+  ALT,
+
+  /**
+   An unspecified other alternate allele. This may occur when the variant for
+   the genotype partially encoded by this allele is split as a multi-allelic site.
+   */
+  OTHER_ALT,
+
+  /**
+   Missing allele.
+   */
+  NO_CALL
+}
+
+/**
+ Genotype.
+ */
+record Genotype {
+
+  /**
+   The variant for this genotype.
+   */
+  union { null, Variant } variant = null;
+
+  /**
+   The reference contig the variant for this genotype exists on. VCF column 1 "CONTIG".
+   */
+  union { null, string } contigName = null;
+
+  /**
+   The zero-based start position of the variant for this genotype on the reference contig.
+   VCF column 2 "POS" converted to zero-based coordinate system, closed-open intervals.
+   */
+  union { null, long } start = null;
+
+  /**
+   The zero-based, exclusive end position of the variant for this genotype on the reference
+   contig. Calculated by Variant.start + Variant.referenceAllele.length().
+   */
+  union { null, long } end = null;
+
+  /**
+   Sample identifier for this genotype. Join with Sample.sampleId for sample metadata.
+   VCF header line sample id.
+   */
+  union { null, string } sampleId = null;
+
+  /**
+   One or more allele values encoding this genotype. If this genotype is phased, the order of
+   the allele values is significant. VCF genotype field reserved key "GT" mapped to Allele enum
+   values.
+   */
+  array<Allele> alleles = [];
+
+  /**
+   True if this genotype is phased.  If true, the order of alleles is significant. VCF genotype field
+   reserved key "GT" allele separators: '|' genotype is phased; '/' genotype is unphased.
+   */
+  union { boolean, null } phased = false;
+}
+
+/**
+ Genotype annotation.
+ */
+record GenotypeAnnotation {
```

Comment:
I prefer having `GenotypeAnnotation` nest in `Genotype` than the other way around. The reason I like nesting `Variant` in `VariantAnnotation` is because `Variant` nests in `Genotype`, and thus doing a join between `Genotype`s and `VariantAnnotation`s makes sense to me. Historically, we had `VariantCallingAnnotations` as a nested record solely for convenience. I.e., for a "finished" genotype dataset like 1KG, you would probably not populate the `VariantCallingAnnotations`, since they're just used to determine whether a genotype call is correct or not, and the "finished" callset should be "correct".